src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,557 @@
+            Tropospheric model.
+        delay_type (Union[str, Literal["wet", "dry", "hydrostatic", "comb"]]):
+            Type of tropospheric delay.
+        reference_file (List[Filename]): List of reference files.
+        secondary_file (List[Filename]): List of secondary files.
+        interferogram (str): Interferogram identifier.
+        reference_time (datetime.datetime): Reference time.
+        secondary_time (datetime.datetime): Secondary time.
+        delay_datacube (Optional[np.ndarray]): Optional delay data cube.
+
+    """
+
+    x_coordinates: np.ndarray
+    y_coordinates: np.ndarray
+    z_coordinates: np.ndarray
+    bounds: Bbox
+    shape: tuple[int, int]
+    epsg: int
+    geotransform: list[float]
+    wavelength: float
+    tropo_model: Union[
+        str, Literal["ERA5", "HRES", "ERAINT", "ERAI", "MERRA", "NARR", "HRRR", "GMAO"]
+    ]
+    delay_type: Union[str, Literal["wet", "dry", "hydrostatic", "comb"]]
+    reference_file: list[Filename]
+    secondary_file: list[Filename]
+    interferogram: str
+    reference_time: datetime.datetime
+    secondary_time: datetime.datetime
+    delay_datacube: Optional[np.ndarray] = None
+
+
+def estimate_tropospheric_delay(
+    ifg_file_list: list[Path],
+    slc_files: dict[tuple[datetime.datetime], list[Filename]],
+    troposphere_files: dict[tuple[datetime.datetime], list[Filename]],
+    geom_files: list[Path],
+    dem_file: Optional[Path],
+    output_dir: Path,
+    tropo_package: str,
+    tropo_model: str,
+    tropo_delay_type: str,
+    strides: dict[str, int] = {"x": 1, "y": 1},
+):
+    """Estimate the tropospheric delay corrections for each interferogram.
+
+    Parameters
+    ----------
+    ifg_file_list : list[Path]
+        List of interferogram files.
+    slc_files : Dict[datetime.date, list[Filename]]
+        Dictionary of SLC files indexed by date.
+    troposphere_files : Dict[datetime.date, list[Filename]]
+        Dictionary of troposphere files indexed by date.
+    geom_files : list[Path]
+        List of geometry files.
+    dem_file : Optional[Path]
+        DEM file.
+    output_dir : Path
+        Output directory.
+    tropo_package : str
+        Troposphere processing package ('pyaps' or 'raider').
+    tropo_model : str
+        Tropospheric model (ERA5, HRES, ...).
+    tropo_delay_type : str
+        Tropospheric delay type ('wet', 'hydrostatic', 'comb').
+    strides : Dict[str, int], optional
+        Strides for resampling, by default {"x": 1, "y": 1}.
+    """
+    # Read geogrid data
+    xsize, ysize = io.get_raster_xysize(ifg_file_list[0])
+    crs = io.get_raster_crs(ifg_file_list[0])
+    gt = io.get_raster_gt(ifg_file_list[0])
+    ycoord, xcoord = oput.create_yx_arrays(gt, (ysize, xsize))  # 500 m spacing
+    epsg = crs.to_epsg()
+    out_bounds = io.get_raster_bounds(ifg_file_list[0])
+
+    # prepare geometry data
+    logger.info("Prepare geometry files...")
+    geometry_dir = output_dir / "../../geometry"
```

Comment:
Should we just use `output` dir here? I can't tell what the assumed directory layout is, but if we're passing
```
        out_dir = cfg.work_directory / cfg.correction_options._tropo_directory
```
, then the two-directories-up level is above `work_directory`. We can't assume that we have access to write, since SDS often gives very particular permissions that limit our writing to `work_directory`